### PR TITLE
MekView updates

### PR DESCRIPTION
**Pull request by SJuliez**
Some updates to the unit summary:

- adds a toolbar like on the AS Card that allows font configuration and copying the info as text or html
- removes some background coloring that wasnt compatible with dark look and feel, now damage is colored in the configurable warning color aka red and caution color aka yellow
- scrolling should be smoother (smaller increment)
- the tables (armor, weapons etc) now have a little space between columns, most columns are centered
- unnecessary empty lines between text blocks have been removed
- html tags in the fluff text are removed for text export, so simple html tags like `<P>, <BR>, <B> and <I>` can be used to add line breaks and formatting
- the MUL link from the "Source:" text has been removed, there's now a button
- instead, when the source is Shrapnel, a link to the Shrapnel website on bg.battletech.com is present
- in the lobby, the summary scales correctly according to GUI scaling (override on SetVisible)

image

### CODE REVIEW

**Review by rjhancock — APPROVED** [review]
Sanity check looked fine to me.

File: megamek/src/megamek/common/MechView.java [hunk 1]
```diff
@@ -261,18 +282,16 @@ public MechView(final Entity entity, final boolean showDetail, final boolean use
...
         sHead.add(new LabeledElement(Messages.getString("MechView.Cost"),
                 dFormatter.format(cost) + " C-bills"));
-        if (entity.hasMulId()) {
-            sHead.add(new HyperLinkElement(MMConstants.MUL_URL_PREFIX + entity.getMulId(),
-                    Messages.getString("MechView.Source")));
-            if (!entity.getSource().isEmpty()) {
-                sHead.add(new LabeledElement("", entity.getSource()));
+        String source = entity.getSource();
+        if (!source.isBlank()) {
+            if (source.contains("Shrapnel")) {
```
[the remark this answers is not in the document]
Comment (NickAragua) [reply megamek/src/megamek/common/MechView.java:287]: Also, not sure if we really want a magic string here.

File: megamek/src/megamek/common/MechView.java [hunk 2]
```diff
@@ -1184,12 +1212,16 @@ private static class LabeledElement implements ViewElement {
         
         @Override
         public String toPlainText() {
-            return label + ": " + value + "\n";
+            String htmlCleanedText = value.replaceAll("<[Bb][Rr]> *", "\n")
```
Comment (NickAragua) [inline megamek/src/megamek/common/MechView.java:1215]: Is that really all the html tags that can show up in this text?

**Review by NickAragua — CHANGES_REQUESTED** [review]
One question, one minor change request.

File: megamek/src/megamek/common/MechView.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/common/MechView.java:287]: Well the source is just a text in the unit files. If it says "shrapnel" or "Shranpel" then it's misspelled and waiting for correction :). This feels like something for the canon units, not player constructed. I'm not against using toLowerCase. Just doesn't look like it needs versatility.
On magic string: What do you suggest? make it a constant?

image

[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/src/megamek/common/MechView.java:287]: If we had a list of the products (like from MUL) could we have them in a separate file to manage? I think I have one.

[the remark this answers is not in the document]
Comment (NickAragua) [reply megamek/src/megamek/common/MechView.java:287]: Yeah, I think a constant will do for now.

File: megamek/src/megamek/common/MechView.java (hunk 2 above)
Comment (NickAragua) [reply megamek/src/megamek/common/MechView.java:1215]: Ok for now. If we wind up having html elements that aren't getting caught then we can look into a more robust solution.

**Review by NickAragua — APPROVED** [review]

**Pull request merged**